mvrx-navigation/src/main/kotlin/com/airbnb/mvrx/navigation/navigationLifecycleAwareLazy.kt

```diff
@@ -0,0 +1,94 @@
+@file:Suppress("ClassName")
+
+package com.airbnb.mvrx.navigation
+
+import androidx.annotation.RestrictTo
+import androidx.annotation.VisibleForTesting
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import java.io.Serializable
+
+private object UninitializedValue
+
+/**
+ * This was copied from SynchronizedLazyImpl but modified to automatically initialize in ON_CREATE.
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY)
+@SuppressWarnings("Detekt.ClassNaming")
+class navigationLifecycleAwareLazy<out T>(
+    owner: LifecycleOwner,
+    initializer: () -> T
+) : Lazy<T>, Serializable {
+    private var initializer: (() -> T)? = initializer
+
+    @Volatile
+    @SuppressWarnings("Detekt.VariableNaming")
+    private var _value: Any? = UninitializedValue
+
+    // final field is required to enable safe publication of constructed instance
+    private val lock = this
+
+    @VisibleForTesting
+    internal val lifecycleObserver: DefaultLifecycleObserver =
```

Comment:
You changed this from the original impl and marked it as `@VisibleForTesting` but I don't see any tests for it. If you aren't adding new tests, can you keep the impl as close as possible so it can be more easily diffed?

Reply:
Ah, yer. If there were any tests I've since removed them. We can match the original impl more here.